src/observable/forkJoin.ts

```diff
@@ -7,16 +7,15 @@ import {isArray} from '../util/isArray';
 
 export class ForkJoinObservable<T> extends Observable<T> {
   constructor(private sources: Array<Observable<any> | Promise<any>>,
-              private resultSelector?: (...values: Array<any>) => any) {
+              private resultSelector?: (...values: Array<any>) => T) {
```

Comment:
`T` isn't used anywhere in the sources of the selector, to `T` is really just the result.  In this case it would return a string based on the code snippet.